packages/gestalt/src/TypeaheadInputField.js

```diff
@@ -0,0 +1,207 @@
+
+  const handleChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    onChange({
+      value: event.currentTarget.value,
+      event,
+    });
+  };
+
+  const handleClear = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    onChange({ value: '', event });
+
+    if (onClear) onClear();
+  };
+  const handleMouseEnter = () => setHovered(true);
+
+  const handleMouseLeave = () => setHovered(false);
+
+  const handleFocus = (event: SyntheticFocusEvent<HTMLInputElement>) => {
+    if (onFocus) {
+      onFocus({
+        value: event.currentTarget.value,
+        event,
+      });
+    }
+  };
+
+  const handleBlur = (
+    event:
+      | SyntheticFocusEvent<HTMLInputElement>
+      | SyntheticKeyboardEvent<HTMLInputElement>
+  ) => {
+    if (onBlur) {
+      onBlur({ event });
+    }
+  };
+
+  const handleClick = (event: SyntheticFocusEvent<HTMLInputElement>) => {
+    handleFocus(event);
+    setContainer(true);
+  };
+
+  const handleKeyNavigation = (
+    event: SyntheticKeyboardEvent<HTMLInputElement>
+  ) => {
+    setContainer(true);
+    // Up Arrow
+    if (event.keyCode === 38) {
+      onKeyNavigation(-1);
+    }
+    // Down Arrow
+    else if (event.keyCode === 40) {
+      onKeyNavigation(1);
+    }
+    // Enter Key
+    else if (event.keyCode === 13) {
+      handleBlur(event);
+    }
+  };
+
+  const hasValue = value && value?.length > 0;
+
+  const className = classnames(styles.input, {
+    [layout.medium]: size === 'md',
+    [layout.large]: size === 'lg',
+    [styles.inputActive]: true,
+    [styles.inputHovered]: hovered,
+    [typeaheadStyle.inputRadius]: true,
+  });
+
+  const clearButtonSize = size === 'lg' ? 24 : 20;
+  const clearIconSize = size === 'lg' ? 12 : 10;
+
+  return (
+    <>
+      {label && <FormLabel id={id} label={label} />}
+      <Box
+        alignItems="center"
+        display="flex"
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}
+        position="relative"
+      >
+        <input
+          ref={forwardedRef}
+          autoComplete="off"
+          aria-label={label}
+          className={className}
+          id={id}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onClick={() => setContainer(true)}
+          onChange={handleChange}
+          placeholder={placeholder}
+          type="text"
+          value={value}
+          onKeyDown={handleKeyNavigation}
+        />
+
+        <button
+          className={hasValue ? styles.clear : typeaheadStyle.icon}
+          onClick={!hasValue ? handleClick : handleClear}
+          tabIndex={-1}
+          type="button"
+        >
+          <Box
+            alignItems="center"
+            color="transparent"
+            display="flex"
+            height={clearButtonSize}
+            justifyContent="center"
+            rounding="circle"
+            width={clearButtonSize}
+          >
+            <Icon
+              accessibilityLabel="typeahead clear button"
```

Comment:
@simkessy since this is not selectable by keyboard, could you change it to an empty accessibility label? `accessibilityLabel=""`